docs/SQL/development/visual-editor/variant-schema.md

```diff
@@ -0,0 +1,85 @@
+
+![Schema and column selector](img/variant-infer-schema.png)
+
+:::note
+
+The inferred schema is cached so that you can use it again in the future whenever you reopen the Model, Gem, or another Gem connected to the same input port. Check the bottom of the column selector to see the last time the variant schema was inferred.
+
+To refresh the schema, simply click **Infer Schema** again.
+
+:::
+
+After you infer the schema, you can click **Edit Schema** to view the variant schema and make edits to it. Use the Type dropdowns to manually choose the data type of each nested schema.
+
+![Edit schema view](img/variant-edit-schema.png)
+
+You can always use the Code view to write the code query if you want to set the schema yourself.
+
+## Variant sampling setting
+
+When Prophecy infers the variant schema, it samples the records to identify all potential iterations of keys and values within the schema.
+
+The default number of records that Prophecy parses to understand the nested data schema is 100. You can update this limit under the Development Settings, which you can navigate to by clicking **... > Development Settings**.
+
+![Variant sampling setting](img/variant-sampling-setting.png)
+
+We recommend that you increase the limit for small structures, or decrease it for larger ones.
+
+:::note
+
+This setting does not rely on the ratio of the data since that would require a complete count of the data records.
+
+:::
+
+## Adding a nested column
+
+With in the column selector, you can add a nested column by clicking **Add Column** next to the input field name.
+
+![Add column](img/variant-add-column.png)
+
+When adding a column nested within a variant, the output column name, expression, and data type are automatically generated according to the following rules:
+
+- Column name: The column name matches the input field name. If there's a duplicate, the name is prefixed with the next available parent field until it becomes unique.
+
+  For example, if the column name `name` already exists, the new field might be named `customers_name`.
+
+- Expression: The expression represents the full path to the selected field, and uses existing flattened subpaths.
+
+- Data type: The data type is automatically `CAST` to the closest inferred type.
+
+### Default casting
+
+Prophecy automatically adds a `CAST` to any column you add from a nested type. By default, the column is cast using the standard `CAST(x AS y)` syntax.
+
+In some cases, a path within a variant may hold different value types across rows. For instance, consider a Dataset where each row’s value key contains different data types, such as integer, object, and boolean.
+
+Prophecy supports this scenario by presenting each detected data type for a given key, array, or object as a separate item in the column selector. When you add one of those columns to the expression, instead of explicit casting we use the `as_*` function, which returns `null` if the cast is not possible.
```

Comment:
We actually use cast() API to perform the cast by default. So if cast is not possible, it actually errors out.
User can change this behaviour by setting try_cast.

Reply:
Done.